src/jit/jitdump.cc

```diff
@@ -0,0 +1,215 @@
+#include "jit/jitdump.hpp"
+
+#include "logger.hpp"
+#include "span.hpp"
+
+#include <fstream>
+#include <vector>
+
+// If we want to consider big endian, we will need this
+// has bswap_64/32/16
+// #include <byteswap.h>
+
+namespace ddprof {
+
+namespace {
+uint64_t load64(const uint64_t *data) {
+  uint64_t Ret;
+  memcpy(&Ret, data, sizeof(uint64_t));
+  // #ifdef BIG_ENDIAN
+  //   bswap_64(Ret);
+  // #endif
+  return Ret;
+}
+
+int32_t load32(const int32_t *data) {
+  int32_t Ret;
+  memcpy(&Ret, data, sizeof(int32_t));
+  // #ifdef BIG_ENDIAN
+  //   bswap_32(Ret);
+  // #endif
+  return Ret;
+}
+
+} // namespace
+
+DDRes jit_read_header(std::ifstream &file_stream, JITHeader &header) {
+  file_stream.read(reinterpret_cast<char *>(&header), sizeof(JITHeader));
+  if (!file_stream.good()) {
+    DDRES_RETURN_ERROR_LOG(DD_WHAT_JIT, "incomplete jit file");
+  }
+  if (header.magic == k_header_magic) {
+    // expected value (no need to swap data)
+  } else if (header.magic == k_header_magic_rev) {
+    // todo everything should be swapped throughout the parsing (not handled)
+    DDRES_RETURN_ERROR_LOG(DD_WHAT_JIT, "Swap data not handled");
+  } else {
+    DDRES_RETURN_ERROR_LOG(DD_WHAT_JIT, "Unknown jit format");
+  }
+  auto remaining_size = header.total_size - sizeof(header);
+  if (remaining_size) {
+    std::vector<char> read_buf;
+    read_buf.resize(remaining_size);
+    file_stream.read(read_buf.data(), remaining_size);
+  }
+  if (header.version != k_jit_header_version) {
+    DDRES_RETURN_ERROR_LOG(DD_WHAT_JIT, "Version not handled");
+  }
+  return ddres_init();
+}
+
+// true if we should continue
+bool jit_read_prefix(std::ifstream &file_stream, JITRecordPrefix &prefix) {
+  file_stream.read(reinterpret_cast<char *>(&prefix), sizeof(JITRecordPrefix));
+  if (!file_stream.good()) {
+    // It is expected that we reach EOF here
+    return false;
+  }
+  if (prefix.id == JITRecordType::JIT_CODE_CLOSE) {
+    return false;
+  }
+  if (prefix.id >= JIT_CODE_MAX) {
+    LG_WRN("Unknown JIT Prefix ID");
+    return false;
+  }
+  return true;
+}
+
+DDRes jit_read_code_load(std::ifstream &file_stream,
+                         JITRecordCodeLoad &code_load,
+                         std::vector<char> &buff) {
+#ifdef DEBUG
+  LG_DBG("----  Read code load  ----");
+#endif
+  buff.resize(code_load.prefix.total_size - sizeof(JITRecordPrefix));
```

Comment:
if `code_load.prefix.total_size < sizeof(JITRecordPrefix)`, this will be incorrect